[Deps] Update `automattic-encryptedlogging` to 1.1.1

ParaskP7 opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

Closes: [AINFRA-1281](https://linear.app/a8c/issue/AINFRA-1281)

---

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR updates automattic-encryptedlogging to [1.1.1](https://github.com/Automattic/EncryptedLogging/releases/tag/1.1.1).

FYI: This update resolves a long standing `OutOfMemory` (OOM) error when using this library alongside logs that are of large size.

PS: This change also resolved a regression that was caused because of #14407, which brought-in some app logs improvements (9892c31cde40d49684101550cf60194aa49871d4). For more info see (internal discussion): p1759832867149029-slack-CGPNUU63E (See also: [EL#29](https://github.com/Automattic/EncryptedLogging/issues/29))

---

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

1. Using `App Inspection` from AS find the `encrypted-log.db` and its sole `EncryptedLogEntity` table. Notice that the table is empty, there are no log to be sent.
2. Prepare the app for a crash. Navigate to `Menu` -> `Settings` -> `Privacy settings` and then switch-on the `Report Crashes` option.
3. Using the patch below, make the app crash. You could navigate to `Menu` -> `Settings` -> `Developer Options` and then click `Crash The App`.

<details>
    <summary>patch</summary>

```diff
Subject: [PATCH] Build: Remove unnecessary encrypted logging proguard rules

The 'automattic-logging' library defines these Proguard rules already,
via its 'consumer-rules.pro' file, and thus those are automatically 
applies to any module that uses this dependency.

consumer-rules.pro: https://github.com/Automattic/EncryptedLogging/blob/
trunk/encryptedlogging/consumer-rules.pro
---
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/developer/DeveloperOptionsViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/developer/DeveloperOptionsViewModel.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/developer/DeveloperOptionsViewModel.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/developer/DeveloperOptionsViewModel.kt	(revision 2b8d77c042fff8f0df4629edf9a0e84e1961b97f)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/developer/DeveloperOptionsViewModel.kt	(date 1752838777405)
@@ -90,12 +90,25 @@
         )
     )
 
+    private val crashTheApp = flowOf(
+        NonToggleableListItem(
+            icon = R.drawable.ic_cancel,
+            iconTint = R.color.color_primary,
+            label = UiString.UiStringText("Crash The App"),
+            isEnabled = true,
+            onClick = {
+                throw Exception("Developer caused a crash!")
+            }
+        )
+    )
+
     val viewState = combine(
         simulatedCardReaderFlow,
         readerUpdateFrequencyFlow,
         interacPaymentEnabledFlow,
         savedPrivacySettingsOnDialogFlow,
-        apiFakerFlow
+        apiFakerFlow,
+        crashTheApp
     ) { items ->
         DeveloperOptionsViewState(
             rows = items.filterNotNull()
```

</details>

4. You might now notice a new db entry within `App Inspection` from AS on that `DETACHED` app (or not, it depends). Now, switch to the `ATTACHED` app and notice the db empty. This is because this db entry/log has been already sent to Sentry already, on app start and after the app crashed. To really notice the entry, trying crashing the app again, this time put the device on offline mode first. Doing that you'll be able to see the entry on the db, even when switching to the newly created `ATTACHED` app.
5. Open [Sentry](https://a8c.sentry.io) and navigate to `Issues`. Find `woocommerce-android` and check the latest `debug` exceptions. You could search for `Developer caused a crash!` and check the latest crash. Now navigate at the bottom and check the `uuid`. Verify it is the same `UUID` that you saw on that db entry before.
6. Open [Read Encrypted Logs](https://mc.a8c.com/encrypted-logs.php), copy-paste that `UUID` into `Log UUID`, click `View` and make sure you can read the newly uploaded and now decrypted log.
7. Finally, you could inspect `Logcat` and notice/verify the below entries:
    - Success: `Successfully uploaded encrypted log with uuid: <UUID>`
    - Failure: `Failed to upload encrypted log with uuid: <UUID> [Reason: <ERROR>]`

> [!TIP]
> FYI: If you want to be thorough about testing this, you could follow the "extra" testing instructions on [EL#28](https://github.com/Automattic/EncryptedLogging/pull/28). The testing instruction in there focus on how to replicate an OOM, filling the logs and asserting that this fix works no matter how big (in size) the to-be encrypted/uploaded log file ends-up being.

---

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
Fix: Revert enqueue sending encrypted logs back to blocking

FYI: This is done because we've identified a regression where the actual
encrypted log, although seen as sent, is not actually ending-up being
uploaded into our systems. Thus, when someone then tries to browse it,
based on it UUID, they will get: "Log File Not Found.Invalid Log File"

For more info see: p1759832867149029-slack-CGPNUU63E

PS: The PR that introduced this regression is #14407, which brought-in
some app logs improvements.

ParaskP7 committed:
Deps: Update automattic-encryptedlogging to 1.1.1

Release Notes: https://github.com/Automattic/EncryptedLogging/releases/
tag/1.1.1

FYI: This update resolves a long standing OutOfMemory (OOM) error when
using this library alongside logs that are of large size.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 commented on WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/EnqueueSendingEncryptedLogs.kt:
```diff
@@ -3,23 +3,20 @@ package com.woocommerce.android.util.crashlogging
         eventLevel: EventLevel
     ) {
-        appCoroutineScope.launch {
+        runBlocking {
             encryptedLogging.enqueueSendingEncryptedLogs(
                 uuid = uuid,
                 file = encryptedLogsFileProvider.provide(),
```
Yes, we know and agreed on that, for more info see (internal discussion): p1759832867149029-slack-CGPNUU63E

ParaskP7 commented on WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/EnqueueSendingEncryptedLogs.kt:
```diff
@@ -3,23 +3,20 @@ package com.woocommerce.android.util.crashlogging
     private val networkStatus: NetworkStatus,
-    @AppCoroutineScope private val appCoroutineScope: CoroutineScope
 ) {
     operator fun invoke(
         uuid: String,
         eventLevel: EventLevel
     ) {
```
Yes, we know and agreed on that, for more info see (internal discussion): p1759832867149029-slack-CGPNUU63E

ParaskP7 committed:
Test: Update enqueue sending encrypted logs test

FYI: Because of this fix: 9892c31cde40d49684101550cf60194aa49871d4

ParaskP7 committed:
Fix: Enqueue sending encrypted logs as non-blocking for not-fatal errors

FYI: After some discussions we agreed on enqueuing the logs on the
'main' thread for 'FATAL' error, else do this off the 'main' thread.

ParaskP7 commented on WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/EnqueueSendingEncryptedLogs.kt:
```diff
@@ -7,24 +7,34 @@ import com.woocommerce.android.di.AppCoroutineScope
-        appCoroutineScope.launch {
+        if (eventLevel == FATAL) {
+            val logs = runBlocking { encryptedLogsFileProvider.provide() }
             encryptedLogging.enqueueSendingEncryptedLogs(
                 uuid = uuid,
-                file = encryptedLogsFileProvider.provide(),
-                shouldUploadImmediately = eventLevel != FATAL && networkStatus.isConnected()
+                file = logs,
+                shouldUploadImmediately = false
             )
+        } else {
```
Yes, we know and agreed on that, for more info see (internal discussion): p1759832867149029-slack-CGPNUU63E

ParaskP7 commented on WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/EnqueueSendingEncryptedLogs.kt:
```diff
@@ -7,24 +7,34 @@ import com.woocommerce.android.di.AppCoroutineScope
 import com.woocommerce.android.tools.NetworkStatus
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
```
Yes, we know and agreed on that, for more info see (internal discussion): p1759832867149029-slack-CGPNUU63E

hichamboushaba reviewed (approved):
Works as expected, nice work @ParaskP7, thanks for the additional improvements too 🙏 

Just a FYI, I deleted the test crash reports in Sentry.
